### ISSUE TITLE
Explore: accepting completion item removes following text

### ISSUE THREAD

**Issue by davkal** [grafana/grafana#13885]
Reported to me by @tomwilkie 

0. Explore with a Prometheus datasource
1. Enter query `up`
2. Put cursor to front and add `su`
3. Accept the completion item function `sum`

Result: `sum(|)`
Expected in order of desirability:
```
sum(|up)
sum(|)up
sum|up
```

CC @miqh

**Comment by miqh** [comment]
@davkal, because you've flagged it as a bug: did this ever work as per the described expected behaviour? I've a feeling it never did.

What are your thoughts on leveraging the `PlaceholdersBuffer` to support this?

Also, how smart _should_ the query modification have to be? Things could get out of hand.

For example, would it simply be a matter of always enclosing the current contents of the query editor with the selected typeahead suggestion, even for multi-parameter functions?

Some sample cases:

```promql
1. metric| > pick topk() > topk(metric)|
2. metric| > pick topk() > topk(|, metric) > topk(3|, metric)
3. to| > pick topk() > topk(|, ) > topk(3|, ) > press TAB > topk(3, |) > topk(3, metric|)
```

**Comment by davkal** [comment]
You're right, it never worked that way. I must have dreamt this 🔮 

My concern is that it should be robust and non-destructive first. We can then experiment with placeholder logic after the base case is covered and fully tested.

I consider the base case to be: `to|metric` => pick `topk` => `topk|metric`

When in doubt, I play with VSCode's intellisense. It's quite popular, and I believe being consistent with popular products is beneficial.

**Comment by davkal** [comment]
Fixed via #13954 by @miqh
